modelimplementation/src/main/java/org/apache/pdfbox/cos/COSDocument.java

```diff
@@ -0,0 +1,652 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.pdfbox.cos;
```

Comment:
Patches to PDF Box should be in a separate,dedicated project. There are three reasons for this:

 * it decouples the PDF Box parser from the library, we may use another parser in the future;
 * it allows us to submit patch requests to PDFBox if we desire; and
 * it avoids incompatible source licences as Apache licensed code can't be distributed with GPL.